### PR TITLE
Limit animations to specific properties

### PR DESCRIPTION
**Pull request by ConfusedVorlon**
as things stand, buttons sometimes animate to their starting positions because the button style is providing an animation which is then used by the system for the position.

see this video for an example:
https://www.dropbox.com/s/6am1lkcz79g6xzv/current_animation.mov?dl=0

when the sheet appears - the button animates to the centre spot.

specifying the animation value in elevation.swift fixes this
(essentially telling swiftUI that this animation only applies to changes of elevation

see fixed version:
https://www.dropbox.com/s/gq8ni7ktsm82c2a/fixed_animation.mov?dl=0

I have added specific values everywhere there is an animation as that seems like a good practice to keep the animations encapsulated. I haven't tested on tvos. Also - it's not my code, so please do review and check that I have correctly identified the correct single value which is changing to drive the animations.

### CODE REVIEW

File: Sources/MaterialUI/DisabledFader.swift [hunk 1]
```diff
@@ -15,6 +15,6 @@ struct DisabledFader: ViewModifier {
 		content
 			.compositingGroup()
 			.opacity(isEnabled ? 1 : 0.5)
-			.animation(nil)
+            .animation(nil,value:isEnabled)
```
Comment (chkn) [inline Sources/MaterialUI/DisabledFader.swift]:
```suggestion
			.animation(nil, value: isEnabled)
```

File: Sources/MaterialUI/Elevation.swift [hunk 2]
```diff
@@ -53,15 +53,15 @@ fileprivate struct Elevation: ViewModifier {
 	public func body(content: Content) -> some View
 	{
 		content
-			.animation(nil)
+            .animation(nil,value:elevation)
```
Comment (chkn) [inline Sources/MaterialUI/Elevation.swift]:
```suggestion
			.animation(nil, value: elevation)
```

File: Sources/MaterialUI/Elevation.swift [hunk 3]
```diff
@@ -53,15 +53,15 @@ fileprivate struct Elevation: ViewModifier {
...
 			// top shadow
 			.shadow(color: Color.black.opacity(Self.kAmbientShadowOpacity), radius: ambientShadowBlur, x: 0, y: 0)
 
 			// key shadow
 			.shadow(color: Color.black.opacity(Self.kKeyShadowOpacity), radius: keyShadowBlur, x: 0, y: keyShadowYOff)
 
 			// FIXME: Is this the correct curve?
-			.animation(isEnabled ? Animation.easeInOut(duration: Self.kThemeChangeDuration) : nil)
+            .animation(isEnabled ? Animation.easeInOut(duration: Self.kThemeChangeDuration) : nil,value:elevation)
```
Comment (chkn) [inline Sources/MaterialUI/Elevation.swift]:
```suggestion
			.animation(isEnabled ? Animation.easeInOut(duration: Self.kThemeChangeDuration) : nil, value: elevation)
```

File: Sources/MaterialUI/Ripple.swift [hunk 4]
```diff
@@ -60,7 +60,7 @@ fileprivate struct Ripple: ViewModifier {
 			.fixedSize(horizontal: !isRadial, vertical: !isRadial)
 			.offset(rippleOffset(mX, mY))
 			.scaleEffect(rippleAppeared ? 1 : Self.kRippleStartingScale)
-			.animation(.easeInOut(duration: isRadial ? Self.kRadialReactionDuration : Self.kRippleTouchDownDuration))
+            .animation(.easeInOut(duration: isRadial ? Self.kRadialReactionDuration : Self.kRippleTouchDownDuration),value:rippleAppeared)
```
Comment (chkn) [inline Sources/MaterialUI/Ripple.swift]:
```suggestion
			.animation(.easeInOut(duration: isRadial ? Self.kRadialReactionDuration : Self.kRippleTouchDownDuration), value: rippleAppeared)
```

File: Sources/MaterialUI/Ripple.swift [hunk 5]
```diff
@@ -73,14 +73,14 @@ fileprivate struct Ripple: ViewModifier {
 						RoundedRectangle(cornerRadius: cornerRadius)
 							.fill(color.opacity(pointerState.isOver ? 0.08 : 0))
 							.frame(width: dimension, height: dimension)
-							.animation(.linear(duration: Self.kRippleFadeInOutDuration))
+                            .animation(.linear(duration: Self.kRippleFadeInOutDuration),value:pointerState.isOver)
```
Comment (chkn) [inline Sources/MaterialUI/Ripple.swift]:
```suggestion
							.animation(.linear(duration: Self.kRippleFadeInOutDuration), value: pointerState.isOver)
```

File: Sources/MaterialUI/Ripple.swift [hunk 6]
```diff
@@ -73,14 +73,14 @@ fileprivate struct Ripple: ViewModifier {
...
 					}
 					// ripple
 					if showRipple {
 						RoundedRectangle(cornerRadius: cornerRadius)
 							.fill(color.opacity(0.16))
 							.frame(width: dimension, height: dimension)
-							.animation(.linear(duration: Self.kRippleFadeInOutDuration))
+							.animation(.linear(duration: Self.kRippleFadeInOutDuration),value:pointerState.isOver)
```
Comment (chkn) [inline Sources/MaterialUI/Ripple.swift]:
```suggestion
							.animation(.linear(duration: Self.kRippleFadeInOutDuration), value: pointerState.isOver)
```

File: Sources/MaterialUI/Toggle.swift [hunk 7]
```diff
@@ -41,12 +41,12 @@ public struct MaterialSwitchToggleStyle: ToggleStyle {
 				.fill(configuration.isOn ? Color.accentColor : Color.white)
 				.frame(width: Self.thumbDiameter, height: Self.thumbDiameter)
 				.offset(offset)
-				.animation(anim)
+                .animation(anim,value:configuration.isOn)
```
Comment (chkn) [inline Sources/MaterialUI/Toggle.swift]:
```suggestion
				.animation(anim, value: configuration.isOn)
```

File: Sources/MaterialUI/Toggle.swift [hunk 8]
```diff
@@ -41,12 +41,12 @@ public struct MaterialSwitchToggleStyle: ToggleStyle {
...
 				.elevation(2)
 			}
 			.modifier(DisabledFader())
 			.radialRipple(configuration.isOn ? Color.accentColor : Color.primary, offset: offset)
-			.animation(anim)
+			.animation(anim,value:configuration.isOn)
```
Comment (chkn) [inline Sources/MaterialUI/Toggle.swift:51]:
```suggestion
			.animation(anim, value: configuration.isOn)
```

**Review by chkn — COMMENTED** [review]
Good catch finding that animation bug! I think you got all the right values- at least, all the animations still appear correct to me. Just a few code style nitpick issues and then we can merge this.

**Comment by ConfusedVorlon** [comment]
I think this is the first time I have actually hit tabs vs spaces in real life :)
also - confirming MIT licence is fine.

**Review by chkn — APPROVED** [review]
This looks good, thanks for making those changes! Looks like it just has some conflicts now after merging your other PR. Once those are fixed this can be merged

**Comment by ConfusedVorlon** [comment]
updated

**Comment by chkn** [comment]
And merged! Thanks again!

**Comment by ConfusedVorlon** [comment]
Thanks for being so quick and involved. It's refreshing when pull requests are not ignored!

**Pull request merged**